app/Http/Middleware/SetCacheHeadersIfAuth.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Ushahidi\App\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Carbon;
+use Illuminate\Contracts\Auth\Factory as Auth;
+
+class SetCacheHeadersIfAuth
+{
+    /**
+     * The authentication guard factory instance.
+     *
+     * @var \Illuminate\Contracts\Auth\Factory
+     */
+    protected $auth;
+
+    /**
+     * Create a new middleware instance.
+     *
+     * @param  \Illuminate\Contracts\Auth\Factory  $auth
+     * @return void
+     */
+    public function __construct(Auth $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    /**
+     * Add cache related HTTP headers.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  string  $ifAuth
+     * @param  string|array  $options
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function handle($request, Closure $next, $ifAuth = null, $options = [])
+    {
+        $response = $next($request);
+
+        if ($ifAuth != null) {
+            if ($this->auth->guard($ifAuth)->guest()) {
```

Comment:
we can probably combine/move up this condiditon to the first one no?

Reply:
yep :+1: